common/scripted_effects/wc_mana_effects.txt

```diff
@@ -0,0 +1,62 @@
+﻿wc_regenerate_mana_effect = {
+	if = {
+		limit = { has_variable = wc_current_mana_variable }
+
+		change_variable = { name = wc_current_mana_variable add = wc_mana_regeneration_value }
+		wc_recalculate_mana_effect = yes
+	}
+	else = { set_variable = { name = wc_current_mana_variable value = wc_max_mana_value } }
+
+	wc_recalculate_prowess_from_mana_effect = yes
+}
+wc_restore_AMOUNT_of_mana = {
+	if = {
+		limit = { has_variable = wc_current_mana_variable }
+
+		change_variable = { name = wc_current_mana_variable add = $AMOUNT$ }
+		wc_recalculate_mana_effect = yes
+	}
+	else = { set_variable = { name = wc_current_mana_variable value = wc_max_mana_value } }
+
+	wc_recalculate_prowess_from_mana_effect = yes	
+}
+wc_recalculate_mana_effect = {
+	if = {
+		limit = { var:wc_current_mana_variable > wc_max_mana_value }
+
+		set_variable = { name = wc_current_mana_variable value = wc_max_mana_value }
+	}
+}
+
+wc_recalculate_prowess_from_mana_effect = {
+	remove_all_character_modifier_instances = wc_prowess_from_available_mana_modifier
+
+	while = {
+		count = wc_current_mana_to_prowess_value
+
+		add_character_modifier = { modifier = wc_prowess_from_available_mana_modifier }
+	}
+}
+
+# wc_spend_X_mana_effect = { change_variable = { name = wc_current_mana_variable add = $VALUE$ } }
+
+wc_start_mana_regeneration_loop_effect = {
+	if = {
+		limit = {
+			NOT = { has_variable = wc_current_mana_variable }
+			wc_max_mana_value > 0
+		}
+
+		wc_regenerate_mana_effect = yes
+		trigger_event = {
+			on_action = wc_on_mana_regeneration
+			days = { 1 28 } # So it will not cause a major lag on every 1st of each month
+		}
+	}
+}
+wc_fill_mana_on_day_one_from_perks_crutch_effect = { # Crutch for mages to have full mana on day 1
+	if = {
+		limit = { game_start_day_trigger = yes }
+		set_variable = { name = wc_current_mana_variable value = wc_max_mana_value }
```

Comment:
You set the mana without `wc_recalculate_mana_effect` and `wc_recalculate_prowess_from_mana_effect`.
The most right and clear way to fix it will be creating change current mana and set current mana effects that will handle prowess change, overflows, etc.
The overflow should be fixed by `clamp_variable`.

Reply:
> You set the mana without `wc_recalculate_mana_effect` and `wc_recalculate_prowess_from_mana_effect`.

Fixed?

> The overflow should be fixed by `clamp_variable`.

I tried, but it didn't fix it.